### PR TITLE
json: support custom error handler

### PR DESCRIPTION
**Pull request by wuhan005**

### CODE REVIEW

File: binding.go [hunk 1]
```diff
@@ -13,11 +13,22 @@ import (
 	"github.com/flamego/flamego"
 )
 
+// Options is a struct for specifying configuration options for the binding middleware.
+type Options struct {
+	// CustomErrorHandler will be invoked if errors occurred.
+	CustomErrorHandler func(flamego.Context, Errors)
```
Comment (unknwon) [inline binding.go:19]:
```suggestion
	// ErrorHandler will be invoked when errors occurred.
	ErrorHandler func(flamego.Context, Errors)
```

We may use this field to define default error handler.

**Review by unknwon — CHANGES_REQUESTED** [review]

File: binding.go [hunk 2]
```diff
@@ -37,6 +48,11 @@ func JSON(model interface{}) flamego.Handler {
 			}
 		}
 		validateAndMap(c, validate, obj, errs)
+
+		errs = c.Value(reflect.TypeOf(errs)).Interface().(Errors)
+		if len(errs) > 0 && option.ErrorHandler != nil {
+			_, _ = c.Invoke(option.ErrorHandler)
```
Comment (unknwon) [inline binding.go:54]: Should we panic on error here? Swallow error would cause confusion

Comment (wuhan005) [reply binding.go]: Fixed.

**Review by unknwon — APPROVED** [review]

**Pull request merged**